app/controllers/application_controller.rb

```diff
@@ -28,7 +29,9 @@ def append_info_to_payload(payload)
   attr_writer :analytics
 
   def analytics
-    @analytics ||= Analytics.new(current_user, request)
+    user = current_user || AnonymousUser.new
```

Comment:
We can look into it in a separate PR, but it won't work in this PR without additional code because some tests will break and we'll need to add additional methods to the AnonymousUser class that aren't related to this PR. Also, we use Devise's `user_signed_in?` in a few places, and that will always return `true` if we do override `current_user` like this:
```ruby
def current_user
  @current_user = super || AnonymousUser.new
end
```
TL;DR This is not a trivial change and is not related to this PR.